### ISSUE TITLE
[FEATURE] Enhance SPDK library to support snapshot chain management

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#5823]
## Is your feature request related to a problem? Please describe (👍 if you like this request)

The current implementation of the SPDK library only provides information about logical volume (lvol)'s parent and children lvols and doesn't support the management of the snapshot chain. This results in longhorn-spdk-engine actively and periodically syncing with the lvols information from SPDK library and maintaining a snapshot chain for a volume, which can cause time delay and extra resource consumption.

To make the management of lvols easier and more efficient, it would be helpful to enhance the SPDK library to support snapshot chain management. This would involve implementing the management of snapshots (logical volumes) within the SPDK library. By doing so, longhorn-spdk-engine could directly interact with SPDK library to manage snapshot chains instead of periodically syncing information.

## Describe the solution you'd like

A clear and concise description of what you want to happen

**Comment by derekbit** [comment]
> https://spdk.io/doc/logical_volumes.html#lvol_rpc

I mean the management of the snapshot chain/tree which is consist of lvols.

**Comment by l-mb** [comment]
> > https://spdk.io/doc/logical_volumes.html#lvol_rpc
> 
> I mean the management of the snapshot chain/tree which is consist of lvols.

Yes, management is mentioned in the original description. I was wondering what this means in more detail - what calls/functions are missing from the current API?

**Comment by innobead** [comment]
cc @longhorn/dev-data-plane

**Comment by DamiaSan** [comment]
@derekbit 
Is https://github.com/longhorn/spdk/pull/16 what you mean by snapshot management?

**Comment by shuo-wu** [comment]
@DamiaSan 
No. Currently, there is no such a SPDK API that can directly retrieve the whole snapshot chain (bottom up) or the snapshot tree (top down) with one given lvol name. Our Longhorn SPDK server has to periodically list all lvols and construct the snapshot tree for each replica.
